packages/aws-cdk-lib/aws-lambda/lib/function.ts

```diff
@@ -1283,6 +1293,40 @@ Environment variables can be marked for removal when used in Lambda@Edge by sett
     };
   }
 
+  private configureSnapStart(props: FunctionProps): CfnFunction.SnapStartProperty | undefined {
+    // return/exit if no snapStart included
+    if (!props.snapStart) {
+      return undefined;
+    }
+
+    // SnapStart does not support arm64 architecture, Amazon Elastic File System (Amazon EFS), or ephemeral storage greater than 512 MB.
+    // SnapStart doesn't support provisioned concurrency either, but that's configured at the version level,
+    // so it can't be checked at function set up time
+    // SnapStart supports the Java 11 and Java 17 (java11 and java17) managed runtimes.
+    // See https://docs.aws.amazon.com/lambda/latest/dg/snapstart.html
+    if (props.snapStart != SnapStartConfig.ON_PUBLISHED_VERSIONS) {
```

Comment:
Instead of checking the entire object, you should check if `props.snapStart.ApplyOn  != 'PublishedVersions'`

This compares the full objects, and customers should be able to configure this as 
`snapStart: new SnapStartConfig('PublishedVersions');`
if they want instead of always having to use the `SnapStartConfig.ON_PUBLISHED_VERSIONS`.

Reply:
For we moved to use abstract class/render we can't check the value inside `applyOn` So currently I'll stick with this approach. Thanks